android-json-form-wizard/src/main/java/com/vijay/jsonwizard/fragments/JsonFormFragment.java

```diff
@@ -225,19 +229,15 @@ public void setmJsonApi(JsonApi mJsonApi) {
     @Override
     public void onResume() {
         super.onResume();
-
-        if (!skipBlankSteps()) { return; }
-
-        if (!getJsonApi().isPreviousPressed()) {
-            skipStepsOnNextPressed();
-        } else {
```

Comment:
@bennsimon any idea why the skip blank step logic still doesn't work for `JsonFormFragment`s? Seems related to this code being removed and moved to JsonWizardFormFragmentPresenter https://github.com/OpenSRP/opensrp-client-native-form/pull/407/files#diff-026e337bf5459fc5eee13061b659364e67ba948390ac996edef6fb3feae6ca3dR53.

Reply:
Blank steps are not being skipped on both back and next press on the `JsonFormFragment`.

Reply:
can i see your form?

Reply:
check this out https://github.com/OpenSRP/opensrp-client-native-form/blob/b6bdad0041abc719e125725fd53efb4325705d99/android-json-form-wizard/src/main/java/com/vijay/jsonwizard/presenters/JsonWizardFormFragmentPresenter.java#L43 ...

Reply:
Ok, can we move this call back to onResume? https://github.com/OpenSRP/opensrp-client-native-form/blob/b6bdad0041abc719e125725fd53efb4325705d99/android-json-form-wizard/src/main/java/com/vijay/jsonwizard/presenters/JsonWizardFormFragmentPresenter.java#L53

Reply:
and would that solve the backpress issues as well?

Reply:
`JsonFormFragmentPresenter` has `onNextClick` you can copy the changes from `JsonWizardFormFragmentPresenter` ... 

**Ok, can we move this call back to onResume?**
No dont move this

Reply:
Check this too https://github.com/OpenSRP/opensrp-client-native-form/blob/abe02b5e496ed17dbfbb48cb569bd257b12480c0/android-json-form-wizard/src/main/java/com/vijay/jsonwizard/presenters/JsonFormFragmentPresenter.java#L286

Reply:
Moving to `JsonFormFragmentPresenter` doesn't seem to work. Could you take a look? https://github.com/OpenSRP/opensrp-client-native-form/pull/512